stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -43,9 +47,17 @@
    * server streaming calls, but may receive many onNext callbacks.  Servers may invoke onNext at
    * most once for client streaming calls, but may receive many onNext callbacks.
    *
-   * <p>If an exception is thrown by an implementation the caller is expected to terminate the
-   * stream by calling {@link #onError(Throwable)} with the caught exception prior to
-   * propagating it.
+   * <p>Although implementations are expected to not throw exceptions, if an exception occurs the
+   * caller is encouraged to call {@link #onError} as part of normal application exception handling.
```

Comment:
This sounds like a subtle change to the contract, specifically that callers are no longer expected to necessarily pass the caught exception itself to `onError`?